app/models/gemmy.rb

```diff
@@ -9,6 +9,17 @@ class Gemmy < ApplicationRecord
 
   delegate :to_param, :to_s, to: :name
 
+  # Find existing by case-insensitive name
+  def self.find_by_name(name, raise_error: false)
+    find_by!("LOWER(name) = ?", name.downcase)
+  rescue ActiveRecord::RecordNotFound => e
+    raise e if raise_error
+  end
+
+  def self.find_by_name!(name)
+    find_by_name(name, raise_error: true)
+  end
```

Comment:
...and added these two methods for case-insensitive search.

The other option of course would be to change the data type to `citext`, which requires a postgres extension. I think that would be cleaner but didn't want to go quite that far.